src/main/java/com/aws/greengrass/status/FleetStatusService.java

```diff
@@ -337,21 +337,24 @@ private void updatePeriodicFleetStatusData() {
     }
 
     /**
-     * Update the Fleet Status information for all the components.
-     * @param isConfigurationUpdate true if the update is triggered by device configuration changes
-     *                              false if the update is triggered at kernel launch IoTJobsHelper post-inject
+     * Trigger a Fleet Status update at kernel launch.
      */
-    public void updateFleetStatusUpdateForAllComponents(Boolean isConfigurationUpdate) {
-        if (isConfigurationUpdate) {
-            updateFleetStatusUpdateForAllComponents(Trigger.NETWORK_RECONFIGURE);
+    public void triggerFleetStatusUpdateAtKernelLaunch() {
+        if (!deviceConfiguration.isDeviceConfiguredToTalkToCloud()) {
+            logger.atWarn().log("Failed to send status update at kernel launch because device is offline");
```

Comment:
should we be warning about this? Seems like it would be expected behavior to not make network calls when device is not configured with cloud endpoints

I suggest dropping to DEBUG and saying something like: "Will not send status update for Nucleus Launch as device is not configured for online access"

Reply:
As we discussed offline, will keep it at WARN level to keep parity with other device offline message.